workflow/controller/operator.go

```diff
@@ -2011,6 +2018,7 @@ func (woc *wfOperationCtx) initializeCacheNode(nodeName string, resolvedTmpl *wf
 	woc.log.Debug("Initializing cached node ", nodeName, common.GetTemplateHolderString(orgTmpl), boundaryID)
 	node := woc.initializeExecutableNode(nodeName, wfutil.GetNodeType(resolvedTmpl), templateScope, resolvedTmpl, orgTmpl, boundaryID, wfv1.NodePending, messages...)
 	node.MemoizationStatus = memStat
+	woc.recordNodePhaseEventOfPhase(node, wfv1.NodeRunning)
```

Comment:
I really need you to avoid sprinkling this statement throughout the code. I'd put money on this creating us a game of whack-o-mole. Can you centralize all events to `markPhase`

Reply:
Good point. We need to capture both creation in running state and transitions directly from lending to complete state. Let me take another look.

Reply:
Updated in https://github.com/argoproj/argo-workflows/pull/5531/commits/92a8abb066695f7c547ca7bb6ef0f32bb8210294 per your suggestion below to use `persistUpdates` as a central update location.